ark/utils/segmentation_utils_test.py

```diff
@@ -137,3 +161,32 @@ def test_transform_expression_matrix_multiple_compartments():
     for cell in cell_data.cell_id:
         arcsinh_vals = np.arcsinh(cell_data.loc[:, cell, modified_cols].values)
         assert np.array_equal(arcsinh_data.loc[:, cell, modified_cols].values, arcsinh_vals)
+
+
+def test_visualize_watershed():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        channel_xr = _generate_channel_xr()
+        overlay_channels = channel_xr.channels.values[:2],
+        segmentation_labels_xr = _generate_deepcell_ouput()
+
+        segmentation_utils.visualize_watershed_transform(
+            segmentation_labels_xr=segmentation_labels_xr,
+            fovs=segmentation_labels_xr.fovs, channel_data_xr=channel_xr,
+            overlay_channels=overlay_channels,
+            output_dir=temp_dir)
+        for mod_output_fov in segmentation_labels_xr.fovs:
+            for chan_list in overlay_channels:
+                assert os.path.exists(
+                    os.path.join(temp_dir, '_'.join(
+                        [f'{mod_output_fov.values}', *chan_list, 'overlay.tiff'])))
+
+        segmentation_utils.visualize_watershed_transform(
+            segmentation_labels_xr=segmentation_labels_xr,
+            fovs=segmentation_labels_xr.fovs, channel_data_xr=channel_xr,
+            overlay_channels=overlay_channels,
+            output_dir=temp_dir, save_tifs='all')
+        for mod_output_fov in segmentation_labels_xr.fovs:
+            assert os.path.exists(
+                os.path.join(temp_dir, "{}_segmentation_borders.tiff".format(mod_output_fov)))
+            assert os.path.exists(
+                os.path.join(temp_dir, "{}_segmentation_labels.tiff".format(mod_output_fov)))
```

Comment:
I personally would use a separate temp_dir for this. Doesn't look like any of the names conflict with each other, but it helps to ensure a clean slate for tests that are doing separate things. Easier to identify errors then.

Reply:
basically, just add another `with tempfile.TemporaryDirectory() as temp_dir:` on line 182; no need for nested temp dirs